elementary/monitor/data_monitoring/data_monitoring_report.py

```diff
@@ -103,22 +103,23 @@ def generate_report(
             models = self.models_api.get_models(exclude_elementary_models)
             sources = self.models_api.get_sources()
             exposures = self.models_api.get_exposures()
+
             models_runs = self.models_api.get_models_runs(
                 days_back=days_back, exclude_elementary_models=exclude_elementary_models
             )
-            tests_metadata = self.tests_api.get_tests_metadata(days_back=days_back)
 
+            test_results_db_rows = self.tests_api.get_all_test_results_db_rows(
+                days_back=days_back,
+                invocations_per_test=test_runs_amount,
+                disable_passed_test_metrics=disable_passed_test_metrics,
+            )
             (
                 test_results,
                 test_results_totals,
                 invocation,
-            ) = self._get_test_results_and_totals(
-                days_back=days_back,
-                disable_passed_test_metrics=disable_passed_test_metrics,
-            )
+            ) = self._get_test_results_and_totals(test_results_db_rows)
```

Comment:
Can we maybe change those to return objects and not tuples?
(I know it was like that beforehand.)

Reply:
Done :)